tests/e2e/preupgrade_datasetup.go

```diff
@@ -0,0 +1,186 @@
+	// 2. Create PVC-1, PVC-2, PVC-3 which uses the StorageClass created in step 1.
+	// 3. Wait for PV-1, PV-2, PV-3  to be provisioned.
+	// 4. Wait for PVC's status to become Bound..
+	// 6. Create pods using PVC's on specific node.
+	// 7. Wait for Disk to be attached to the node.
+
+	ginkgo.It("[csi-block-vanilla-preupgradedata] Verify creation of dynamic pvc and pods", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		scParameters := make(map[string]string)
+		scParameters[scParamFsType] = ext4FSType
+		// Create Storage class and PVC
+		ginkgo.By("Creating Storage Class and PVC with allowVolumeExpansion = true")
+		var err error
+		storageClassName = "preupgrade-sc-pods"
+
+		namespace, err := framework.CreateTestingNS(f.BaseName, client, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		scSpec := getVSphereStorageClassSpec(storageClassName, scParameters, nil, "", "", true)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("storage class created is : %s", sc.Name)
+
+		count := 0
+		for count < 3 {
+			ginkgo.By("Creating PVCs using the Storage Class")
+			pvclaims[count], err = fpv.CreatePVC(client, namespace.Name,
+				getPersistentVolumeClaimSpecWithStorageClass(namespace.Name, diskSize, sc, nil, ""))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			count++
+		}
+		ginkgo.By("Waiting for all claims to be in bound state")
+		persistentvolumes, err = fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		podCount := 0
+		for podCount < 3 {
+			ginkgo.By("Creating pod to attach PVs to the node")
+			pvclaim = pvclaims[podCount]
+			podArray[podCount], err = createPod(client, namespace.Name, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			podCount++
+		}
+
+		var vmUUID string
+		ginkgo.By("Verify the volumes are attached to the node vm")
+		podCount = 0
+		for podCount < 3 {
+			pvclaim = pvclaims[podCount]
+			pv := getPvFromClaim(client, namespace.Name, pvclaim.Name)
+
+			volumeID := pv.Spec.CSI.VolumeHandle
+			if vanillaCluster {
+				vmUUID = getNodeUUID(ctx, client, podArray[podCount].Spec.NodeName)
+			}
+			ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s",
+				pv.Spec.CSI.VolumeHandle, podArray[podCount].Spec.NodeName))
+			isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volumeID, vmUUID)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskAttached).To(gomega.BeTrue(),
+				fmt.Sprintf("Volume: %s is not attached to the node: %s",
+					pv.Spec.CSI.VolumeHandle, podArray[podCount].Spec.NodeName))
+			podCount++
+		}
+
+		for _, pv := range persistentvolumes {
+			framework.Logf("Volume: %s should be present in the CNS",
+				pv.Spec.CSI.VolumeHandle)
```

Comment:
here where we are checking CNS entry for Pod/PVC/PV?

I can see you have put this CNS log message but I do not see CNS verification happening anywhere.